localization/autoware_gyro_odometer/CMakeLists.txt

```diff
@@ -9,6 +9,10 @@ ament_auto_add_library(${PROJECT_NAME} SHARED
   src/diagnostics_module.cpp
 )
 
+include_directories(
+  src
+)
+
```

Comment:
Why are these lines necessary?

Reply:
It seems when changing the directory structure, the compiler can not find the header in this case.

For the original structure case,
```
src/
    gyro_odometer/
        gyro_odometer_core.hpp
```
and the including `#include "gyro_odometer/gyro_odometer_core.hpp"` use the relative path for the header and this make compiler to search the `gyro_odometer_code.hpp` with searching in relative path 
 `gyro_odometer/gyro_odometer_core.hpp`.

But for this PR, I changed the dir structure to `src/gyro_odometer_core.hpp` and the including `#include "gyro_odometer_core.hpp"`. This will result in compiler to search the `gyro_odometer_core.hpp` in same dir, i.e. `test/gyro_odometer_core.hpp`.

ref: https://cmake.org/cmake/help/latest/command/target_include_directories.html

The
```
include_directories(
  src
)
```
itself might not be appropriate which uses `include_directories` , I will make a commit to use `target_include_directories` instead.